docs/system_overview/configuration.md

```diff
@@ -14,254 +14,676 @@ This scrambled password file is saved after an `iinit` is run. If this file does
+    // Contains subtle network and password related variables.
+    // These values should be changed only in concert with all connecting clients and other servers in the Zone.
+    "advanced_settings": {
+        // The number of threads enabled when parallel transfer is invoked.
+        "default_number_of_transfer_threads": 4,
+
+        // The number of seconds a server-side temporary password is good.
+        "default_temporary_password_lifetime_in_seconds": 120,
+
+        // (Optional)
+        // A list of iRODS server hostnames the delay server dispatches rules to for execution.
+        // If the list is empty, the delay server will dispatch rules to the local server for execution.
+        "delay_rule_executors": [],
+
+        // (Optional)
+        // The number of seconds the delay server sleeps before checking the catalog for rules
+        // ready for execution.
+        "delay_server_sleep_time_in_seconds": 30,
+
+        // (Optional)
+        // Contains settings for controlling the behavior of the internal DNS cache.
+        "dns_cache": {
+            // The amount of shared memory allocated to the DNS cache.
+            "shared_memory_size_in_bytes": 5000000,
+
+            // The amount of time an entry stays valid before being evicted from the cache.
+            "eviction_age_in_seconds": 3600
+        },
 
-  - `db_odbc_type` (required) - The ODBC type, usually 'unix'
+        // (Optional)
+        // Contains settings for controlling the behavior of the internal Hostname cache.
+        "hostname_cache": {
+            // The amount of shared memory allocated to the Hostname cache.
+            "shared_memory_size_in_bytes": 2500000,
 
-  - `db_password` (required) - The password for the `db_username` to connect to the `db_name`
+            // The amount of time an entry stays valid before being evicted from the cache.
+            "eviction_age_in_seconds": 3600
+        },
 
-  - `db_port` (required) - The port on which the database server is listening
+        // TODO
+        "maximum_size_for_single_buffer_in_megabytes": 32,
+
+        // (Optional)
+        // The maximum number of bytes available to the delay queue.
+        // When set to 0, the delay server will use as much memory as it needs to hold queued rules.
+        "maximum_size_of_delay_queue_in_bytes": 0,
+
+        // TODO
+        "maximum_temporary_password_lifetime_in_seconds": 1000,
+
+        // (Optional)
+        // The number of delay rules allowed to execute simultaneously.
+        "number_of_concurrent_delay_rule_executors": 4,
+
+        // (Optional)
+        // The amount of time the stacktrace file processing thread sleeps before files are
+        // processed and written to the log file.
+        "stacktrace_file_processor_sleep_time_in_seconds": 10,
+
+        // TODO
+        "transfer_buffer_size_for_parallel_transfer_in_megabytes": 4,
+
+        // TODO
+        "transfer_chunk_size_for_parallel_transfer_in_megabytes": 40
```

Comment:
Needs a description.

Is this specific to parallel transfer via a put operation?